Azk agent not running

I have a few running services inside azk vm and a few azk shells started, which I can access.

The http loadbalancer stopped working (connection reset by peer), and azk status says that agent is not running. However, the containers are open and running. Any ideas ? In the last few days I had a lot of problems with azk (last 2 versions, 0.14.4 and 0.15.0), the vm stopped completely on system sleep twice. 

Before, on azk 0.12.1 everything was fine.

Are there any logs I can paste here ?

```
$ azk status
? The agent is not running, would you like to start it? No
azk: azk agent is required but is not running (try `azk agent status`)
```

@fearenales 

Btw, do you know why the test suite gives me this error ? Do I need linux ?

```
$ azk nvm npm test

> azk@0.15.0 test /Users/toni/code/gh/azk
> make test

task: test
/Users/toni/code/gh/azk/bin/azk nvm gulp test  ""
[16:48:30] Using gulpfile ~/code/gh/azk/gulpfile.js
make: *** [test] Error 1
npm ERR! Test failed.  See above for more details.
```

Thanks!

Everything green,except port binding (I already had agent service binding on that port), and file syncing. But I guess the problems are elsewhere. Will investigate later, the main bug didn't happen for 12 hours, so I guess it's ok.

```
  375 passing (2m)
  1 pending
  4 failing

  1) Azk docker module, run method @slow should support bind ports:
     Error: HTTP code is 500 which indicates error: server error - Cannot start container 9d1ebc5102895ccfdc2dbef55883b986b1e7ffdd19d67449baa3fdb4bf46de40: Error starting userland proxy: listen tcp 0.0.0.0:32777: bind: address already in use

      at _stream_readable.js:944:16

2) Azk sync, Worker module should not include content patterns files from except_from option:
     AssertionError: expected '/private/var/folders/f5/bdtz6z3n4ns8mpkp1npyhj6c0000gn/T/azk-test-54492228zn1s/bar/Fred.txt' not to match /bar\/Fred.txt/
      at Test.callee$1$1$ (/azk:0.15.0/spec/sync/worker_spec.js:216:27)

  3) Azk sync, Worker module should exclude the .gitignore content for default:
     AssertionError: expected '/private/var/folders/f5/bdtz6z3n4ns8mpkp1npyhj6c0000gn/T/azk-test-54492n1w3zrk/ignored/Fred.txt' not to match /ignored\/Fred.txt/
      at Test.callee$1$2$ (/azk:0.15.0/spec/sync/worker_spec.js:241:27)

  4) Azk sync, Worker module should exclude the .syncignore content for default in preference to .gitignore:
     AssertionError: expected '/private/var/folders/f5/bdtz6z3n4ns8mpkp1npyhj6c0000gn/T/azk-test-54492cqjv6qo/foo/Moe.txt' to match /ignored\/Fred.txt/
      at Test.callee$1$3$ (/azk:0.15.0/spec/sync/worker_spec.js:267:23)
```

I ran it again and now only one error. It seems it is an intermittent error.

```
1) Azk sync, Worker module should not include content patterns files from except_from option:
     AssertionError: expected '/private/var/folders/f5/bdtz6z3n4ns8mpkp1npyhj6c0000gn/T/azk-test-562383t7d56o/ignored/Fred.txt' not to match /ignored\/Fred.txt/
      at Test.callee$1$1$ (/azk:0.15.0/spec/sync/worker_spec.js:215:27)
```

--- a/src/agent/server.js
+++ b/src/agent/server.js
@@ -38,8 +38,10 @@ var Server = {
       // Load balancer
       yield this.installBalancer();
 
-      // acive docker monitor
-      this._activeDockerMonitor();
+      // active docker monitor
+      if (config('docker:monitor')) {
+        this._activeDockerMonitor();
+      }
 
       log.info_t("commands.agent.started");
       this.starting = false;

--- a/src/config.js
+++ b/src/config.js
@@ -87,6 +87,7 @@ var options = mergeConfig({
       socket          : envs('AZK_DOCKER_SOCKER', "/var/run/docker.sock"),
       host            : new Dynamic("docker:host"),
       namespace       : envs('AZK_NAMESPACE'),
+      monitor         : envs('AZK_DOCKER_MONITOR', os.platform() === 'linux'),
       api_version     : 'v' + envs('AZK_DOCKER_API_VERSION', '1.20'),
       repository      : 'azk',
       default_domain  : 'azk',

